File: ch3/exo3.py
#!/usr/bin/env python3

## calculate the numbers that are above the average of supplied list

def above_average(L):
    total = sum(L)
    average = 0

    if len(L):
        average = total / len(L)
    
    res = [x for x in L if x > average]
    return res 

fred = range(1,1001)
print("fred is: {}".format(above_average(fred)))

File: ch3/exo3.pl
#!/usr/bin/env perl
use strict;
use warnings;
use utf8;

## returns the numbers that are above the average from the list
## of supplied numbers

sub total {
	my $sum = 0;

	foreach (@_) {
		$sum += $_;
	}
	$sum;
}

sub average {
	my $total = total(@_);
	if (@_) {
		return $total / @_;
	}
	else {
		return 0;
	}
}

sub above_average {
	my $average = average(@_);
	my @res;

	foreach (@_) {
		if ($_ > $average) {
			push @res, $_;
		}
	}
	@res;
}

my @fred = above_average(1..1000);
print "\@fred is: @fred\n";
 
